design/ceph/ceph-nvmeof-gateway.md

```diff
@@ -0,0 +1,210 @@
+                                            \       |        /
+                                         +-----------v-----------+
+                                         |     Storage Layer      |
+                                         | +--------------------+ |
+                                         | | OSD 1 (Physical)   | |
+                                         | +--------------------+ |
+                                         +-----------|------------+
+                                                     |
+                                          +----------v----------+
+                                          |   RBD Volume         |
+                                          |   (Block Device)      |
+                                          +----------|------------+
+                                                     |
+                                          +----------v----------+
+                                          | NVMe-oF Target       |
+                                          | (Presents RBD as     |
+                                          | NVMe Namespace)      |
+                                          +----------|------------+
+                                                     |
+                                           +---------v---------+
+                                           |      Network       |
+                                           | TCP/IP or RDMA     |
+                                           | (NVMe-oF Protocol) |
+                                           +---------|-----------+
+                                                     |
+                                      +--------------v--------------+
+                                      |        Client Node          |
+                                      +-----------------------------+
+                                      | NVMe-oF Initiator (nvme-cli)|
+                                      |                             |
+                                      |  Application sees           |
+                                      |  /dev/nvmeXnY               |
+                                      +-----------------------------+
+
+
+
+
+
+* Note:  In this document, when we refer to 'NVMe namespace,' we mean the NVMe protocol's concept of a logical block device, not Kubernetes resource namespaces."
+
+### Components Responsibilities
+
+* **Volume Provisioning**: CSI driver creates RBD image and configures it in NVMe-oF gateway.
+* **Gateway Management**: Rook operator manages gateway lifecycle, scaling, and health.
+* **Load Balancing/k8s Service**: Service provides single endpoint for multiple gateway instances, NVMe namespace exposed over TCP/IP fabric with port 5500.
+* **Client Access**: External clients connect via NVMe-oF initiator to access volumes.
+
+## CephNVMeOFGateway CRD
+
+### Gateway Configuration
+
+- **Group**: `gateway-group-1` - Logical grouping of gateways
```

Comment:
IIUC when multiple CephNVMeOFGateway CRs are created, each group of gateways will work independently and they would have separate load balancing endpoints, right?